skbio/tree/_tree.py

```diff
@@ -651,6 +651,46 @@ def count(self, tips=False):
             return len(list(self.traverse(include_self=True)))
 
     @experimental(as_of="0.4.0")
+    def observed_node_counts(self, tip_counts):
+        """Returns counts of nodes observations from counts of tip observations
+
+        Parameters
+        ----------
+        tip_counts : dict of ints
+            Counts of observations of tips. Keys correspond to tip names in
+            ``self``, and counts are ints greater than zero.
+
+        Returns
+        -------
+        dict
+            Counts of observations of nodes. Keys correspond to node names
+            (internal nodes or tips), and counts are ints greater than zero.
+
+        Raises
+        ------
+        ValueError
+            If a count less than one is observed.
+        MissingNodeError
+            If a count is provided for a tip not in the tree, or for an
+            internal node.
+
+        """
+        result = defaultdict(int)
+        for tip_name, count in tip_counts.items():
+            if count < 1:
```

Comment:
I think it's important to do here, since we don't know who the caller will be (this is a public method). May revisit doing this when I write ``_validate`` for *unifrac.